File: CPP_LAB5/Mattress.cpp
#include<bits/stdc++.h>
#include "Mattress.h"

using namespace std;

// setting data
Mattress::Mattress(string typ, int wid, int len, int deliver, int pric) {
	set_brand(typ);
    width = wid;
    length = len;
    set_price(pric);
    delivery = deliver;
}

// method to print the info
string Mattress::info() {
	string inf = "Mattress " + get_brand() + " " + to_string(width) + " " + to_string(length) + " " + to_string(delivery) + " " + to_string(get_price());
	return inf;
}

// getters

int Mattress::get_width() {
	return width;
}

int Mattress::get_length() {
	return length;
}

int Mattress::get_delivery() {
	return delivery;
}

// comparator exclusive to mattress
bool Mattress::delivery_compare(Mattress obj1, Mattress obj2) {
	if (obj1.get_delivery() == obj2.get_delivery()) {
		return obj1.get_price() < obj2.get_price();
	}
	return (obj1.get_delivery() < obj2.get_delivery());
}

File: IMT2021016_Java_Lab5 (1)/Mattress.java

import java.util.*;
public class Mattress {
    // Data
    private String type;
    private Integer width, length,price, delivery;

    // Constructor
    public Mattress(String typ, int wid, int len, int deliver, int pric) {
        type = typ;
        width = wid;
        length = len;
        price = pric;
        delivery = deliver;
    }

    // Info of the mattress
    public String info() {
        String inf = "Mattress " + type + " " + String.valueOf(width) + " " + String.valueOf(length) + " " + String.valueOf(delivery) + " " + String.valueOf(price);
        return inf;
    }

    // Method to get data
    public String get_type() {
        return type;
    }

    public Integer get_width() {
        return width;
    }

    public Integer get_length() {
        return length;
    }

    public Integer get_price() {
        return price;
    }
    public Integer get_delivery() {
        return delivery;
    }


}
